lib/web/assistant.go

```diff
@@ -291,6 +294,32 @@ func runAssistant(h *Handler, w http.ResponseWriter, r *http.Request, sctx *Sess
 }
 
 func getAssistantClient(ctx context.Context, proxyClient auth.ClientI, sctx *SessionContext) (*ai.Chat, error) {
+	// Try retrieving credentials from the plugin resource first
+	openaiPlugin, err := proxyClient.PluginsClient().GetPlugin(ctx, &pluginsv1.GetPluginRequest{
+		Name:        "openai-default",
+		WithSecrets: true,
+	})
+	if err == nil {
+		creds := openaiPlugin.Credentials.GetBearerToken()
+		if creds == nil {
+			return nil, trace.BadParameter("malformed credentials")
+		}
+		var token string
+		if creds.TokenFile != "" {
+			tokenBytes, err := os.ReadFile(creds.TokenFile)
+			if err != nil {
+				return nil, trace.Wrap(err)
+			}
+			token = strings.TrimSpace(string(tokenBytes))
+		} else {
+			token = creds.Token
+		}
+		client := ai.NewClient(token)
+		chat := client.NewChat(sctx.cfg.User)
+		return chat, nil
+	}
+	log.WithError(err).Debug("Failed to retrieve OpenAI credentials from a plugin resource, will use static config")
```

Comment:
This will always error in OSS and ENT. Can we either skip the "doesn't exist" error or print it only in the cloud?

Reply:
I have removed the log message. Though in case of operational errors, this will make it slightly confusing, as the error that will be logged will be OpenAI complaining about the fallback dummy API key not being valid.